### PR TITLE
DAP 1.42.0 (includes 1.37-1.42)

### LINKED ISSUE #428: Update DAP to 1.42.x
Hello!

Are there any plans to update DAP to 1.41?

### PR DESCRIPTION
**Pull request by jonahgraham**
This PR is to bring LSP4J up to DAP 1.41.0 and fixes #428.

### CODE REVIEW

**Comment by jonahgraham** [comment]
There is a p2 site: https://download.eclipse.org/lsp4j/builds/jonah/dap_1.41/ that contains this PR contents so it can be tested ahead of being merged.

File: org.eclipse.lsp4j.debug/src/main/java/org/eclipse/lsp4j/debug/DebugProtocol.xtend [hunk 1]
```diff
@@ -409,6 +464,117 @@ class CapabilitiesEventArguments {
...
+	 * <p>
+	 * This is an optional property.
+	 */
+	String message;
+	/**
+	 * Optional progress percentage to display (value range: 0 to 100). If omitted no percentage will be shown.
+	 * <p>
+	 * This is an optional property.
+	 */
+	Integer percentage;
```
[the remark this answers is not in the document]
Comment (jonahgraham) [reply org.eclipse.lsp4j.debug/src/main/java/org/eclipse/lsp4j/debug/DebugProtocol.xtend:524]: Thanks - I think you are right, the only `number` fields in the spec are these new ones. the new `requestId` field is listed as a number, but I think that is a mistake: https://github.com/microsoft/debug-adapter-protocol/issues/122

**Comment by apupier** [comment]
@jonahgraham the PR has conflicts. It requires a rebase.

**Comment by jonahgraham** [comment]
Thanks - I will rebase and get this submitted. If we have to make any fixes as spec is changed (see https://github.com/microsoft/debug-adapter-protocol/issues/122) that can be handled in a future release.

**Comment by jonahgraham** [comment]
The branch name isn't correct - 1.41 - but I reused it anyway.

File: org.eclipse.lsp4j.debug/src/main/java/org/eclipse/lsp4j/debug/DebugProtocol.xtend [hunk 2]
```diff
@@ -409,6 +464,155 @@ class CapabilitiesEventArguments {
...
+	 * <p>
+	 * Possible values include - but not limited to those defined in {@link InvalidatedAreas}
+	 */
+	String[] areas;
+	/**
+	 * If specified, the client only needs to refetch data related to this thread.
+	 * <p>
+	 * This is an optional property.
+	 */
+	Double threadId;
```
Comment (jonahgraham) [inline org.eclipse.lsp4j.debug/src/main/java/org/eclipse/lsp4j/debug/DebugProtocol.xtend:606]: This is wrong see https://github.com/microsoft/debug-adapter-protocol/issues/145

Comment (jonahgraham) [reply org.eclipse.lsp4j.debug/src/main/java/org/eclipse/lsp4j/debug/DebugProtocol.xtend]: Subsequent commit to this branch has `threadId` and `frameId` as an `Integer`

**Comment by jonahgraham** [comment]
@KamasamaK I have resolved the previous comments and added 1.42 support (only Invalidate event was added). Please have a review if you have time, otherwise I will submit this very soon as is so that it makes the 0.10.0 release.

**Comment by jonahgraham** [comment]
@KamasamaK if there is anything that you notice I will handle in subsequent PRs.

**Pull request merged**